src/Forms/TurnstileCaptchaField.php

```
@@ -172,71 +175,57 @@ public function validate($validator)
         }
-
-        curl_setopt($ch, CURLOPT_TIMEOUT, 10);
-        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
-        curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, self::config()->verify_ssl);
-        curl_setopt($ch, CURLOPT_POST, true);
-        curl_setopt($ch, CURLOPT_USERAGENT, str_replace(',', '/', 'SilverStripe'));
-        curl_setopt(
-            $ch,
-            CURLOPT_POSTFIELDS,
-            http_build_query([
-                'secret' => $secret_key,
-                'response' => $recaptchaResponse,
-                'remoteip' => $request->getIP(),
-            ])
-        );
-        $response = json_decode(curl_exec($ch), true);
-
-        if (is_array($response)) {
-            $this->verifyResponse = $response;
-
-            if (!array_key_exists('success', $response) || !$response['success']) {
-                $validator->validationError($this->name, _t(
-                    'Terraformers\\TurnstileCaptcha\\Forms\\CaptchaField.EMPTY',
-                    '_Please answer the captcha,
-                     if you do not see the captcha you must enable JavaScript'),
-                    'validation');
-                return false;
-            }
-
-        } else {
+        $curlOptions[CURLOPT_RETURNTRANSFER] = true;
+        $curlOptions[CURLOPT_SSL_VERIFYPEER] = self::config()->verify_ssl;
+        $curlOptions[CURLOPT_USERAGENT] = str_replace(',', '/', 'SilverStripe');
+        $client = HttpClient::create()->getClient();
+
+       try {
+           $response = $client->request('POST', 'https://challenges.cloudflare.com/turnstile/v0/siteverify',[
+               'json' => [
+                   'secret' => $secret_key,
+                   'response' => $recaptchaResponse,
+                   'remoteip' => $request->getIP(),
+               ],
+               'curl' => $curlOptions,
+               'timeout' =>10
+           ]);
+       }catch (Throwable $e) {
+           if($e instanceof RequestException && $e->hasResponse()){
```

this is swallowing quite a few exceptions - instead I think we should add a general validation error if something happens here and log the message using the logger interface (see line 226). So remove this if check, add a validation error here (probably the same as on line 223) and log the error - then return false. (see https://docs.guzzlephp.org/en/stable/quickstart.html#exceptions for the exceptions this is missing)